From 0cff1d9bb695bdc0ad7bad234b92eddf849ce88f Mon Sep 17 00:00:00 2001
From: Mantas Pucka <mantas@8devices.com>
Date: Mon, 2 Jun 2025 17:20:58 +0300
Subject: [PATCH] net: pcs: ipq-uniphy: fix USXGMII link-up failure

USXGMII link-up may fail due to too short delay after PLL reset.
Increase the delay to fix this.

Signed-off-by: Mantas Pucka <mantas@8devices.com>
---
 drivers/net/pcs/pcs-qcom-ipq9574.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/pcs/pcs-qcom-ipq9574.c
+++ b/drivers/net/pcs/pcs-qcom-ipq9574.c
@@ -380,7 +380,7 @@ static int ipq_pcs_config_mode(struct ip
 	if (ret)
 		return ret;
 
-	fsleep(1000);
+	fsleep(20000);
 	ret = regmap_set_bits(qpcs->regmap, PCS_PLL_RESET, PCS_ANA_SW_RESET);
 	if (ret)
 		return ret;
